### PR TITLE
Added new command to download file by section no and refactored commo…

### PR DESCRIPTION
**Pull request by MichaelYoung1981**
…n code.

### CODE REVIEW

File: src/SFA.DAS.QnA.Api/web.config [hunk 1]
```diff
@@ -1,25 +1,31 @@
...
+            <match url="^$" />
+            <conditions>
+              <add input="{HTTP_USER_AGENT}" pattern="^AlwaysOn$" />
+            </conditions>
+            <action type="Rewrite" url="/health" />
+          </rule>
+        </rules>
+      </rewrite>
+    </system.webServer>
+  </location>
```
Comment (MarkFCain) [inline src/SFA.DAS.QnA.Api/web.config]: It should be ok I think, but better make sure the testers know this needs a regression test for EPAO as well as usual tests, as this affect whole app

Comment (MichaelYoung1981) [reply src/SFA.DAS.QnA.Api/web.config]: Change reverted. 👍

File: src/SFA.DAS.QnA.Api/web.config [hunk 2]
```diff
@@ -1,25 +1,31 @@
...
+  <location path="." inheritInChildApplications="false">
+    <system.webServer>
+      <security>
+        <requestFiltering allowDoubleEscaping="true" />
+      </security>
+      <handlers>
+        <add name="aspNetCore" path="*" verb="*" modules="AspNetCoreModuleV2" resourceType="Unspecified" />
+      </handlers>
+      <aspNetCore processPath="%LAUNCHER_PATH%" arguments="%LAUNCHER_ARGS%" stdoutLogEnabled="false" stdoutLogFile=".\logs\stdout" hostingModel="InProcess">
+        <environmentVariables>
```
[the remark this answers is not in the document]
Comment (MichaelYoung1981) [reply src/SFA.DAS.QnA.Api/web.config]: Change reverted. 👍

File: src/SFA.DAS.QnA.Application/Commands/Files/DownloadFile/DownloadFileService.cs [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        internal async Task<HandlerResponse<DownloadFile>> GetDownloadFile(Guid applicationId, ApplicationSection section, string pageId, string questionId, string fileName, CancellationToken cancellationToken)
+        {
+            var page = section.QnAData.Pages.FirstOrDefault(p => p.PageId == pageId);
+
+            if (page == null)
+            {
+                return new HandlerResponse<DownloadFile>(success:false, message:$"Page {pageId} in Application {applicationId} does not exist.");
+            }
+
+            if (page.Questions.All(q => q.Input.Type != "FileUpload"))
```
[the remark this answers is not in the document]
Comment (MichaelYoung1981) [reply src/SFA.DAS.QnA.Application/Commands/Files/DownloadFile/DownloadFileService.cs]: We absolutely should.  Unfortunately that string is hardcoded in a load of places. 😬

**Pull request merged**